### PR TITLE
Api stubs

### PR DESCRIPTION
**Pull request by richafrank**
as well as tooling and docs to generate this for each release

For #1205 

screen shot 2016-05-17 at 10 34 42 am

### CODE REVIEW

File: etc/gen_type_stubs.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import inspect
+from operator import attrgetter
+from textwrap import dedent
+
+from zipline import api, TradingAlgorithm
+
+
+def main():
+    with open(api.__file__ + 'i', 'w') as stub:
```
Comment (llllllllll) [inline etc/gen_type_stubs.py]: we should rstrip 'c' from the filename because it could be `.pyc`

File: docs/source/release-process.rst [hunk 2]
```diff
@@ -27,6 +27,21 @@ update the underline of the title to match the title's width.
 If you are renaming the release at this point, you'll need to git mv the file
 and also update releases.rst to reference the renamed file.
 
+Updating the Python stub files
```
Comment (llllllllll) [inline docs/source/release-process.rst]: should we comment that the imports may need to be updated?

File: etc/gen_type_stubs.py (hunk 1 above)
Comment (richafrank) [inline etc/gen_type_stubs.py]: Agreed.

File: docs/source/release-process.rst (hunk 2 above)
Comment (richafrank) [inline docs/source/release-process.rst]: Good idea.

**Comment by llllllllll** [comment]
lgtm

**Pull request merged**